modules/programs/claude-code.nix

```diff
@@ -156,6 +156,86 @@ in
       };
     };
 
+    hooks = lib.mkOption {
+      type = lib.types.attrsOf lib.types.lines;
+      default = { };
+      description = ''
+        Custom hooks for Claude Code.
+        The attribute name becomes the hook filename, and the value is the hook script content.
+        Hooks are stored in .claude/hooks/ directory.
+      '';
+      example = {
+        pre-edit = ''
+          #!/usr/bin/env bash
+          echo "About to edit file: $1"
+        '';
+        post-commit = ''
+          #!/usr/bin/env bash
+          echo "Committed with message: $1"
+        '';
+      };
+    };
+
+    memory = {
+      text = lib.mkOption {
+        type = lib.types.nullOr lib.types.lines;
+        default = null;
+        description = ''
+          Inline memory content for CLAUDE.md.
+          This option is mutually exclusive with memory.source.
+        '';
+        example = ''
+          # Project Memory
+
+          ## Current Task
+          Implementing enhanced claude-code module for home-manager.
+
+          ## Key Files
+          - claude-code.nix: Main module implementation
+        '';
+      };
+
+      source = lib.mkOption {
+        type = lib.types.nullOr lib.types.path;
+        default = null;
+        description = ''
+          Path to a file containing memory content for CLAUDE.md.
+          This option is mutually exclusive with memory.text.
+        '';
+        example = lib.literalExpression "./claude-memory.md";
+      };
+    };
+
+    agentsDir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        Path to a directory containing agent files for Claude Code.
+        Agent files from this directory will be symlinked to .claude/agents/.
+      '';
+      example = lib.literalExpression ./agents;
+    };
+
+    commandsDir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        Path to a directory containing command files for Claude Code.
+        Command files from this directory will be symlinked to .claude/commands/.
+      '';
+      example = lib.literalExpression ./commands;
+    };
+
+    hooksDir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        Path to a directory containing hook files for Claude Code.
+        Hook files from this directory will be symlinked to .claude/hooks/.
+      '';
+      example = lib.literalExpression ./hooks;
```

Comment:
These dir options remind me that I had the intention of refactoring `sourceFileOrLines` option so I could just change the type of the corresponding options. I have a branch for it, but just haven't gotten to it, yet. Because, I'd rather have the option support either a directory, an attrset of files, or attrset of lines, etc